lib/conduit_async_ssl.ml

```diff
@@ -50,4 +50,9 @@ let ssl_listen ~crt_file ~key_file rd wr =
   >>= fun app_rd ->
```

Comment:
Not being that familiar with the code of conduit I'm wondering if it's possible for `net_to_ssl` to be closed before your cleanup call is made. E.g. after the `app_rd` is created but before `app_wr`. Admittedly, that's a little contrived.